mir/src/rewrite.rs

```diff
@@ -32,7 +32,7 @@ pub fn pull_required_base_columns(q: &mut MirQuery) {
                 !mn.borrow()
                     .ancestors()
                     .iter()
-                    .any(|a| a.borrow().columns().contains(c))
+                    .any(|a| a.borrow().columns().iter().any(|ac|  ac.name == c.name))
```

Comment:
This also relaxes to comparison by column name only, rather than by column name and table name. Check why we changed this.